src/stories/3-fields/button/button.stories.tsx

```diff
@@ -0,0 +1,120 @@
+import * as Icons from "@lifesg/react-icons";
+import { ArgsTable, Description, Heading, PRIMARY_STORY, Stories, Title } from "@storybook/addon-docs";
+import { Meta, StoryFn } from "@storybook/react";
+import { IButtonSchema } from "../../../components/fields/button";
+import { CommonFieldStoryProps, FrontendEngine, OVERRIDES_ARG_TYPE } from "../../common";
+const meta: Meta = {
+	title: "Field/Button",
+	parameters: {
+		docs: {
+			page: () => (
+				<>
+					<Title>Button</Title>
+					<Description>The primary call to action component</Description>
+					<Heading>Props</Heading>
+					<Description>
+						This component also inherits the
+						[HTMLButtonElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement)
+						attributes.
+					</Description>
+					<ArgsTable story={PRIMARY_STORY} />
+					<Stories includePrimary={true} title="Examples" />
+				</>
+			),
+		},
+	},
+	argTypes: {
+		...CommonFieldStoryProps("button"),
+		styleType: {
+			description: "The style type of the button",
+			table: {
+				type: {
+					summary: "default | secondary | light | link",
+				},
+			},
+			options: ["default", "secondary", "light", "link"],
+			control: {
+				type: "select",
+			},
+		},
+		startIcon: {
+			description:
+				"Add <a href='https://designsystem.life.gov.sg/reacticons/index.html?path=/story/collection--page'>Icon</a> to the start of button",
+
+			control: {
+				type: "select",
+			},
+			options: Object.keys(Icons),
+		},
+		endIcon: {
+			description:
+				"Add <a href='https://designsystem.life.gov.sg/reacticons/index.html?path=/story/collection--page'>Icon</a> to the end of button",
+			control: {
+				type: "select",
+			},
+			options: Object.keys(Icons),
+		},
```

Comment:
- add `disabled` argType it is commonly used
- updated `startIcon` and `endIcon` descriptions
```suggestion
		disabled: {
			description: "Specifies if the button is interactable",
			table: {
				type: {
					summary: "boolean | invalid-form",
				},
				defaultValue: { summary: false },
			},
			options: [true, false, "invalid-form"],
			control: {
				type: "select",
			},
		},
		styleType: {
			description: "The style type of the button",
			table: {
				type: {
					summary: "default | secondary | light | link",
				},
			},
			options: ["default", "secondary", "light", "link"],
			control: {
				type: "select",
			},
		},
		startIcon: {
			description:
				"Add an icon based on <a href='https://designsystem.life.gov.sg/reacticons/index.html?path=/story/collection--page' target='_blank' rel='noopener noreferrer'>React Icons</a> before the button label",
			table: {
				type: {
					summary: "Refer to React Icons",
				},
			},
			control: {
				type: "select",
			},
			options: Object.keys(Icons),
		},
		endIcon: {
			description:
				"Add an icon based on <a href='https://designsystem.life.gov.sg/reacticons/index.html?path=/story/collection--page' target='_blank' rel='noopener noreferrer'>React Icons</a> after the button label",
			table: {
				type: {
					summary: "Refer to React Icons",
				},
			},
			control: {
				type: "select",
			},
			options: Object.keys(Icons),
		},
```